pacta/email_is_domain.go

```diff
@@ -0,0 +1,63 @@
+package pacta
+
+// isDomainName checks if a string is a presentation-format domain name. This
+// code comes from an unexported function in Go's 'net' package, see
+// https://github.com/golang/go/blob/09300d89e94da7b1f32742e80768acc711b5c590/src/net/dnsclient.go#L75
+func isDomainName(s string) bool {
```

Comment:
This feels really hard to maintain. Did you write this, or is it sourced elsewhere?

I really don't love having this in our codebase.